### PR TITLE
img-52 Re-package the dataextension classes.

### PR DESCRIPTION
**Pull request by dcruver**
This change moves the data extension header classes into their own package and sets up the library to
complete further, similar refactorings on other NITF elements.

@kcwire 
@bradh

### CODE REVIEW

File: core/src/main/java/org/codice/imaging/nitf/core/common/dataextension/package-info.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ */
+
+/**
+
+ Provides NITF support.
```
Comment (kcwire) [inline core/src/main/java/org/codice/imaging/nitf/core/common/dataextension/package-info.java]: if we're going to keep the package-info files around we should make the javadoc more useful. I personally don't care for them - I'm good with removing it.

File: core/src/main/java/org/codice/imaging/nitf/core/common/package-info.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ */
+
+/**
+
+ Provides NITF support.
```
Comment (kcwire) [inline core/src/main/java/org/codice/imaging/nitf/core/common/package-info.java]: same here.

File: core/src/main/java/org/codice/imaging/nitf/core/common/dataextension/package-info.java (hunk 1 above)
Comment (dcruver) [inline core/src/main/java/org/codice/imaging/nitf/core/common/dataextension/package-info.java]: Removed the files and updated the check style.xml so it doesn't flag it.

File: core/src/main/java/org/codice/imaging/nitf/core/common/package-info.java (hunk 2 above)
Comment (dcruver) [inline core/src/main/java/org/codice/imaging/nitf/core/common/package-info.java]: see previous response.

**Comment by bradh** [comment]
The Travis CI build is failing on OpenJDK 7, but passing on Oracle JDK 7 and 8. The failure looks to be a broken build environment (`*** buffer overflow detected ***: /usr/lib/jvm/java-7-openjdk-amd64/jre/bin/java terminated`) rather than anything related to this code.

Looks like @kcwire comments were addressed, and I'm happy with the change.

**Pull request merged**